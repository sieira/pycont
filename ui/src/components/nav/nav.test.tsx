import React from 'react'
import { createWithProvider } from '../../test-utils/render-utils.test'

import Nav from '.'

it('renders without crashing', () => {
  const navBar = createWithProvider(<Nav />)
  expect(navBar.toJSON()).toMatchSnapshot()
})

it('renders logged in without crashing', () => {
  const navBar = createWithProvider(<Nav />, {
    auth: {
      isAuthenticated: true,
      currentUser: { username: 'Schwarzenegger' },
    },
    accounts: {},
  })
  expect(navBar.toJSON()).toMatchSnapshot()
})
