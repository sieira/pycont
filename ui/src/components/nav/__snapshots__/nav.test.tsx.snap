// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders logged in without crashing 1`] = `
Array [
  <p>
    Auth state: 
    Logged in user: undefined
  </p>,
  <ul>
    <li>
      <a
        aria-current={null}
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/terms"
        onClick={[Function]}
      >
        Terms
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/logout"
        onClick={[Function]}
      >
        Log out
      </a>
    </li>
  </ul>,
]
`;

exports[`renders without crashing 1`] = `
Array [
  <p>
    Auth state: 
    Logged out
  </p>,
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Landing
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/terms"
        onClick={[Function]}
      >
        Terms
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/login"
        onClick={[Function]}
      >
        Log in
      </a>
    </li>
  </ul>,
]
`;
